index.js

```diff
@@ -340,28 +340,36 @@ function fastifyView (fastify, opts, next) {
       }
 
       if (prod) {
-        const html = template(data)
-        if (!this.getHeader('content-type')) {
-          this.header('Content-Type', 'text/html; charset=' + charset)
+        try {
+          const html = template(data)
+          if (!this.getHeader('content-type')) {
+            this.header('Content-Type', 'text/html; charset=' + charset)
+          }
+          this.send(html)
+        } catch (e) {
+          this.send(e)
         }
-        this.send(html)
       } else {
         getPartials(type, options.partials || {}, (err, partialsObject) => {
           if (err) {
             this.send(err)
             return
           }
 
-          Object.keys(partialsObject).forEach((name) => {
-            engine.registerPartial(name, engine.compile(partialsObject[name]))
-          })
+          try {
+            Object.keys(partialsObject).forEach((name) => {
+              engine.registerPartial(name, engine.compile(partialsObject[name]))
+            })
 
-          const html = template(data)
+            const html = template(data)
 
-          if (!this.getHeader('content-type')) {
-            this.header('Content-Type', 'text/html; charset=' + charset)
+            if (!this.getHeader('content-type')) {
+              this.header('Content-Type', 'text/html; charset=' + charset)
+            }
+            this.send(html)
+          } catch (e) {
+            this.send(e)
```

Comment:
this is going to return a JSON, there is not a handlebar feature like a 500 html service page?

Reply:
AFAIK there isn't. The error can be caught in an error handler and you can display a 500 page there.